### PR TITLE
ESLint rules: rules requiring minor changes

### PR DESCRIPTION
**Pull request by JoelMcKinnon**
Added the following rules to ESLint config and resolved issues found (10):

- "no-unused-expressions": "error"
- "no-useless-concat": "error"
- "radix": "error"
- "require-await": "error"
- "no-alert": "error"
- "no-useless-constructor": "error"
- "no-duplicate-imports": "error"

Resolves part of: https://github.com/nasa/openmct/issues/3059

**Testing instructions**

1. Make code changes violating above rules.
2. Run npm run lint.
3. See that violations of rules are observed.
4. Undo changes and see that violations are gone.
(note: violations should also be observed in editor)

**Author Checklist**

- Changes address original issue? Yes
- Unit tests included and/or updated with changes? N/A
- Command line build passes? Yes
- Changes have been smoke-tested? Yes
- Testing instructions included? Yes

### CODE REVIEW

File: src/plugins/notebook/components/notebook-menu-switcher.vue [hunk 1]
```diff
@@ -80,15 +80,15 @@ export default {
                 if (domainObject.location) {
                     const defaultPath = `${domainObject.name} - ${defaultNotebook.section.name} - ${defaultNotebook.page.name}`;
 
-                    notebookTypes.push({
+                    await notebookTypes.push({
```
Comment (shefalijoshi) [inline src/plugins/notebook/components/notebook-menu-switcher.vue:83]: This is odd. Maybe setNotebookTypes shouldn't be async and remove the awaits?

File: src/plugins/timeConductor/ConductorAxis.vue [hunk 2]
```diff
@@ -190,13 +190,23 @@ export default {
 
                 requestAnimationFrame(() => {
                     this.dragX = $event.clientX;
-                    this.inPanMode ? this.pan() : this.zoom();
```
Comment (JoelMcKinnon) [inline src/plugins/timeConductor/ConductorAxis.vue:193]: @davetsay - please check this

File: src/plugins/timeConductor/ConductorAxis.vue [hunk 3]
```diff
@@ -190,13 +190,23 @@ export default {
...
+                        this.zoom()
+                    }
+
                     this.dragging = false;
                 });
             }
         },
         dragEnd() {
-            this.inPanMode ? this.endPan() : this.endZoom();
+            if (this.inPanMode) {
```
Comment (JoelMcKinnon) [inline src/plugins/timeConductor/ConductorAxis.vue:205]: @davetsay - please check this

File: src/plugins/notebook/components/notebook-menu-switcher.vue (hunk 1 above)
Comment (JoelMcKinnon) [reply src/plugins/notebook/components/notebook-menu-switcher.vue:83]: @nikhilmandlik Please take a look at this one. Docs: https://eslint.org/docs/rules/require-await

**Comment by JoelMcKinnon** [comment]
@shefalijoshi I added some await keywords in TelemetryCriterionSpec.js to satisfy require-await rule. Please check.

File: src/plugins/notebook/components/notebook-menu-switcher.vue (hunk 1 above)
Comment (nikhilmandlik) [reply src/plugins/notebook/components/notebook-menu-switcher.vue:83]: Yup, lets remove async from `async setNotebookTypes() {`

**Comment by JoelMcKinnon** [comment]
@shefali - please take a look at TelemetryCriterionSpec.js. I couldn't get the awaits to work so just removed the async. It you think they're needed please advise.

File: src/plugins/notebook/components/notebook-menu-switcher.vue (hunk 1 above)
Comment (JoelMcKinnon) [reply src/plugins/notebook/components/notebook-menu-switcher.vue]: done

File: src/plugins/timeConductor/ConductorAxis.vue (hunk 2 above)
Comment (davetsay) [reply src/plugins/timeConductor/ConductorAxis.vue:193]: LGTM

File: src/plugins/timeConductor/ConductorAxis.vue (hunk 3 above)
Comment (davetsay) [reply src/plugins/timeConductor/ConductorAxis.vue:205]: LGTM

File: src/adapter/services/LegacyObjectAPIInterceptor.js [hunk 4]
```diff
@@ -70,7 +70,7 @@ define([
     };
 
     ObjectServiceProvider.prototype.create = async function (object) {
-        let model = utils.toOldFormat(object);
+        let model = await utils.toOldFormat(object);
```
Comment (davetsay) [inline src/adapter/services/LegacyObjectAPIInterceptor.js:73]: `utils.toOldFormat` is not an async call

File: src/adapter/services/LegacyObjectAPIInterceptor.js [hunk 5]
```diff
@@ -79,7 +79,7 @@ define([
         );
     }
     ObjectServiceProvider.prototype.update = async function (object) {
-        let model = utils.toOldFormat(object);
+        let model = await utils.toOldFormat(object);
```
Comment (davetsay) [inline src/adapter/services/LegacyObjectAPIInterceptor.js:82]: `utils.toOldFormat` is not an async call

Comment (JoelMcKinnon) [reply src/adapter/services/LegacyObjectAPIInterceptor.js:82]: changed to awaiting a variable which then gets returned

File: src/adapter/services/LegacyObjectAPIInterceptor.js (hunk 4 above)
Comment (JoelMcKinnon) [reply src/adapter/services/LegacyObjectAPIInterceptor.js]: changed to awaiting a variable which then gets returned

**Comment by akhenry** [comment]
@davetsay Are you happy with Joel's changes? If so, we have two approved reviews which is more than sufficient for merge.

**Comment by JoelMcKinnon** [comment]
@shefalijoshi - please look at failing tests. Failing in master as well.

**Pull request merged**